### DISCUSSION TITLE
How do I re-build the formula against latest master?

### DISCUSSION THREAD

**Discussion by sankalp-khare** [d12frosted/homebrew-emacs-plus#354] [Q&A] [answered]
My local setup is like:
```
❯ brew info emacs-plus@28
d12frosted/emacs-plus/emacs-plus@28: stable 28.0.50
/usr/local/Cellar/emacs-plus@28/28.0.50 (4,045 files, 133MB) *
  Built from source on 2021-02-20 at 00:22:53 with: --with-modern-black-dragon-icon
From: https://github.com/d12frosted/homebrew-emacs-plus/blob/HEAD/
```

Q: Where does the `28.0.50` in the install location path come from, and if I uninstall + re-install today, will I get a newer version number over there instead?

[The link in the `From:` section](https://github.com/d12frosted/homebrew-emacs-plus/blob/HEAD/) 404s, but I do see that [the master branch of this repo](https://github.com/d12frosted/homebrew-emacs-plus/commits/master) has seen lots of commits since the install date in February 2021.

What is the recommended command / approach to get `brew` to install the newer version of `emacs-plus@28` from the latest master branch of the formula repo, if available. Is this the correct way to think about this?

```
❯ brew install emacs-plus@28 --with-modern-black-dragon-icon
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 2 taps (homebrew/core and homebrew/cask).
==> Updated Formulae
Updated 2 formulae.
==> New Casks
vitalsigns
==> Updated Casks
Updated 6 casks.

Warning: d12frosted/emacs-plus/emacs-plus@28 28.0.50 is already installed and up-to-date.
To reinstall 28.0.50, run:
  brew reinstall emacs-plus@28
```

Should I issue a `reinstall` command as brew suggests above?

Thanks!

**Comment by d12frosted** [comment, marked as answer]
In general, you should not use `reinstall` command even though `brew` suggests so, see [README][1]. 

Version for `emacs-plus@28` is [hard-coded][2], because there is no way brew can infer this as it does with `emacs-plus@27` (see [url][3]) or `emacs-plus@26` (see [url][4]), because Emacs 28 was not released yet, it's a bleeding edge without any release candidates, so you simply build against master.

In order to get the latest version you need to do as readme suggests - uninstall and then install again, which simply pulls the latest changes from Emacs [upstream mirror][5] (see [url in formula][6]). `emacs-plus` repository does not contain sources of Emacs :) It's just a formula.

Hope that helps. 

P. S. Oh nice, I thought they changed UI of issue tracker, but instead you used discussions! 💯 

[1]: https://github.com/d12frosted/homebrew-emacs-plus#reinstall
[2]: https://github.com/d12frosted/homebrew-emacs-plus/blob/7912426a0c74f28438a6e7358f5481a796cd658c/Formula/emacs-plus%4028.rb#L5
[3]: https://github.com/d12frosted/homebrew-emacs-plus/blob/master/Formula/emacs-plus%4027.rb#L5
[4]: https://github.com/d12frosted/homebrew-emacs-plus/blob/master/Formula/emacs-plus%4026.rb#L5
[5]: https://github.com/emacs-mirror/emacs.git
[6]: https://github.com/d12frosted/homebrew-emacs-plus/blob/master/Formula/emacs-plus%4028.rb#L66
